core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -602,9 +667,29 @@ private ExprValue exprAddDateInterval(ExprValue date, ExprValue expr) {
    * @return Date/Datetime resulted from days added to date.
    */
   private ExprValue exprAddDateDays(ExprValue date, ExprValue days) {
-    ExprValue exprValue = new ExprDatetimeValue(date.datetimeValue().plusDays(days.longValue()));
-    return (exprValue.timeValue().toSecondOfDay() == 0 ? new ExprDateValue(exprValue.dateValue())
-        : exprValue);
+    return exprDateApplyDays(date, days.longValue(), true);
+  }
+
+  /**
+   * Adds or subtracts `days` to/from `date`.
+   *
+   * @param date A Date/Time/Datetime/Timestamp value to change.
+   * @param days A days amount to add or subtract.
+   * @param add A flag: true to add, false to subtract.
+   * @return Datetime calculated.
+   */
+  private ExprValue exprDateApplyDays(ExprValue date, Long days, Boolean add) {
+    if (date.type() == DATE) {
+      return new ExprDateValue(add ? date.dateValue().plusDays(days)
+          : date.dateValue().minusDays(days));
+    }
+    var dt = LocalDateTime.MIN;
```

Comment:
you can initialize this using a conditional instead of defaulting it and immediately overriding it.